linetools/isgm/utils.py

```diff
@@ -284,70 +285,56 @@ def complist_from_table(table):
 
     """
     # Convert to QTable to handle units in individual entries more easily
-    table = QTable(table)
+    tkeys = table.keys()
 
     # mandatory and optional columns
-    min_columns = ['RA', 'DEC', 'ion_name', 'z_comp', 'vmin', 'vmax']
+    min_columns = ['RA', 'DEC', 'ion_name', 'z_comp', 'vmin', 'vmax', 'Z', 'ion', 'Ej']
     special_columns = ['name', 'comment', 'logN', 'sig_logN', 'flag_logN']
     for colname in min_columns:
         if colname not in table.keys():
             raise IOError('{} is a mandatory column. Please make sure your input table has it.'.format(colname))
 
-    #loop over rows
+    # Generate components -- Should add to linetools.isgm.utils
     complist = []
-    for row in table:
-        # mandatory
-        coord = SkyCoord(row['RA'].to('deg').value, row['DEC'].to('deg').value, unit='deg')  # RA y DEC must both come with units
-        Zion = name_to_ion(row['ion_name'])
-        zcomp = row['z_comp']
-        vlim =[row['vmin'].to('km/s').value, row['vmax'].to('km/s').value] * u.km / u.s  # units are expected here too
-
-        # special columns
-        try:
-            Ntuple = (row['flag_logN'], row['logN'], row['sig_logN'])  # no units expected
-        except KeyError:
+    coords = SkyCoord(ra=table['RA'], dec=table['DEC'], unit='deg')
+    for kk,row in enumerate(table):
+        if 'flag_logN' in tkeys:
+            Ntuple = (row['flag_logN'], row['logN'], row['sig_logN'])
+        else:
             Ntuple = None
-        try:
-            comment = row['comment']
-        except KeyError:
-            comment = ''
-        try:
-            name = row['name']
-        except KeyError:
-            name = None
-
-        # define the component
-        comp = AbsComponent(coord, Zion, zcomp, vlim, Ntup=Ntuple, comment=comment, name=name)
-
-        # other columns will be filled in comp.attrib dict
-        for colname in table.keys():
-            if (colname not in special_columns) and (colname not in min_columns):
-                kms_cols = ['b', 'sig_b']
-                if colname in kms_cols:  # check units for parameters expected in velocity units
-                    try:
-                        val_aux = row[colname].to('km/s').value * u.km / u.s
-                    except u.UnitConversionError:
-                        raise IOError('If `{}` column is present, it must have velocity units.'.format(colname))
-                    comp.attrib[colname] = val_aux
-                # parameters we do not care about units much
-                else:
-                    comp.attrib[colname] = row[colname]
-
-        # append
-        complist += [comp]
+        abscomp = AbsComponent(coords[kk], (row['Z'], row['ion']),
+                               row['z_comp'], [row['vmin'],row['vmax']]*u.km/u.s,
+                               Ej=row['Ej']/u.cm,
```

Comment:
Chek for units in `vmin, vmax` and `Ej` instead of assuming them.

Reply:
Ok.  Done.